### PR TITLE
Documentation updates and reorganization

### PR DESCRIPTION
**Pull request by asumagic**
## What does this PR do?

Bunch of stuff rewritten, reorganized, simplified and split up. Better watch the diff and final doc result to judge.

<details>
  <summary><b>Before submitting</b></summary>

- [ ] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by asumagic** [comment]
Should be ready to review after a rebase when #2692 is merged assuming no other changes, will open for review at that time.

Note for reviewing: contributing.md etc. were edited but the documentation preview on readthedocs is **not** going to link to that. Check it from GitHub.

**Review by TParcollet — CHANGES_REQUESTED** [review]
LGTM once the very minor title comment is adressed

File: docs/guidance.md
Comment (asumagic) [reply docs/guidance.md:1]: Yeah, I ended up turning this into a simple index for the structure of the project. Some was redundant with other contribution pages and it was just too long-winded IMO.

The only content that was _really_ removed is the `The location of a change foreshadows its integrative complexity.` section which I think was just too complicated. I've tried to reduce the contribution stuff to be as close to the point as possible.

File: docs/contributing.md
Comment (asumagic) [reply docs/contributing.md:16]: I had renamed it because none of the steps are actually specific to SpeechBrain here, so that people with prior experience making PRs could skip it. What do you think?

File: docs/contributing.md
Comment (asumagic) [reply docs/contributing.md:47]: I think contributing.md is already linked to when doing that, but it might be worth making it more obvious.

File: docs/contributing.md
Comment (TParcollet) [reply docs/contributing.md:16]: True.

**Comment by TParcollet** [comment]
@asumagic can you merge develop so that I merge the PR?

**Review by TParcollet — APPROVED** [review]

**Pull request merged**
